br/pkg/lightning/backend/external/file.go

```diff
@@ -54,21 +56,27 @@ func NewKeyValueStore(
 // size or key count exceeds the given distance, a new range property will be
 // appended to the rangePropertiesCollector with current status.
 // `key` must be in strictly ascending order for invocations of a KeyValueStore.
-func (s *KeyValueStore) addEncodedData(val []byte) error {
-	_, err := s.dataWriter.Write(s.ctx, val)
-	if err != nil {
-		return err
+func (s *KeyValueStore) addEncodedData(key []byte, val []byte) error {
+	if len(s.memBuffer) > 5*1024*1024 {
```

Comment:
if we move the `dataWriter.Write` to the end of this function, we don't need to process remaining data in Close. Is it OK?